src/Common/Twig/TwigExtension.php

```diff
@@ -282,7 +282,11 @@ function ($string) {
             new TwigFilter(
                 'dateToTime',
                 function ($str) {
-                    return strtotime($str);
+                    // when there's zeros in the form_vitals entry, it's a date of "00000000000000"
+                    if ($str !== "00000000000000") {
```

Comment:
wondering if instead of handling it like this to add a new function to `SQLUpgradeService.php` that could change these dates to `null`

Reply:
All dates should default to null. Never anything but null. Also if a date datatype is stored as an empty string "" then table date will result in the weird 000000 stuff.

Brady caught most of this when he and a student changed all the date data times in queries.

Reply:
@stephenwaite
You prob should test the $str arg first
`if (($str = strtotime($str)) === false) return  strtotime('now') else return strtotime($str)`

or something like that.

Reply:
I think this is a different situation where the table was already defined with date `null` but data was being inserted with `0000-00-00`.

Reply:
oh, I see, you said that above, thanks @sjpadgett 

> Also if a date datatype is stored as an empty string "" then table date will result in the weird 000000 stuff.

Reply:
you're welcome. any invalid dates in strtotime will pass a false to table and thus 000000000 that's why it's best to validate false to null when stored. hope okay for me to ramble but this gave me a heck of a time in QRDA and CCDA

Reply:
I'm thinking we should have a function in the sql upgrade that will convert the date to the date of the encounter if it's zeroes.

Reply:
Good idea. Might be idea to scan all our important tables and fix dates.
If you need help, let me know and maybe the two of us can come up with something clever. Well at least in our minds!:)